Structure_Engine/Query/CoordinateSystem.cs

```diff
@@ -54,9 +54,9 @@ public static Cartesian CoordinateSystem(this Node node)
         [Output("coordinateSystem", "The local cartesian coordinate system of the Bar.")]
         public static Cartesian CoordinateSystem(this Bar bar)
         {
-            Vector tan = bar.Tangent(true);
-            Vector ax = bar.Normal().CrossProduct(tan);
-            return Engine.Geometry.Create.CartesianCoordinateSystem(bar.StartNode.Position, tan, ax);
+            Vector tan = bar?.Tangent(true);
+            Vector ax = bar?.Normal()?.CrossProduct(tan);
+            return tan != null && ax != null ? Engine.Geometry.Create.CartesianCoordinateSystem(bar.StartNode.Position, tan, ax) : null;
```

Comment:
Having a look through this myself again, hoping to catch some issues to make the review easier. At first I wondered why I had forgot to include error messages at so many points in CoordinateSystem. After looking into it further I realised that this is because Tangent and Normal already use NullCheck and therefore already show error messages. The check in return is just to check whether these null checks failed or passed.